cpp/src/arrow/util/bitmap_ops.cc

```diff
@@ -84,15 +84,26 @@ int64_t CountSetBits(const uint8_t* data, int64_t bit_offset, int64_t length) {
   return count;
 }
 
-enum class TransferMode : bool { Copy, Invert };
+enum class TransferMode : char { Copy, Invert, Revert };
+
+uint8_t revert_uint8(uint8_t num) {
+  num = ((num & 0xf0) >> 4) | ((num & 0x0f) << 4);
+  num = ((num & 0xcc) >> 2) | ((num & 0x33) << 2);
+  num = ((num & 0xaa) >> 1) | ((num & 0x55) << 1);
+  return num;
+}
+
+uint8_t get_reverted_block(uint8_t block_left, uint8_t block_right, uint8_t length) {
+  return revert_uint8(block_left >> (length) | block_right << (8 - length));
+}
 
 template <TransferMode mode>
 void TransferBitmap(const uint8_t* data, int64_t offset, int64_t length,
```

Comment:
Given that the bit-reverse mode has basically 0 in common with the other modes, maybe we can just keep it in an entirely separate function, i.e. use RevertBlockOffsets directly?